### ISSUE TITLE
Switch off PDF attachment to email at Order Confirmation

### ISSUE THREAD

**Issue by NaturePinks** [thirtybees/thirtybees#1156]
The Order Confirmation email to customer always includes the invoice as pdf attachment (if the invoice is available).

It would be a good enhancement to have the ability to switch off attaching the pdf, based on the 'Attach invoice PDF to email' option on the order status. The benefits will be:

1) shave off several seconds for order confirmation (assuming pdf can/will be generated later on demand)
2) avoid duplication of content since the email body has all the details in the invoice

https://forum.thirtybees.com/topic/3792-possible-bug-order-status-attach-invoice-pdf-option-is-not-respected/

**Comment by getdatakick** [comment]
> based on the 'Attach invoice PDF to email' option on the order status

I don't agree with this. Order status email preferences are meant for Order Status Changed email. To reuse the same settings for different email does not make much sense. It would only bring additional questions and problems, for example:

- what if the order status has 'Send an email to the customer' unchecked -- would that mean that 'Order Confirmation' email should not be sent at all?
- Should 'Delivery slip' be attached to the 'Order status' as well, if the checkbox is checked? 
- etc...

If this enhancement should be implemented, then I advocate for much more systematic approach. For example, extract this Order status email settings to generic email level - allow merchants to set this for any email template that exists in the system (and which has order link). Also, different kind of attachments could be supported and automatically added (for example Terms and Conditions pdf, etc)

**Comment by NaturePinks** [comment]
I have looked at all the options for an Order Status and did not see any that will behave oddly when applied at order confirmation too.
For both your questions above, my view is yes and yes. With sane defaults, I can't see anything that goes markedly wrong.

If you don't agree however:
1) how about having a preference on Orders page that allows switching off pdf attachment for order confirmation?
2) how about offloading this option to mail alerts module?

**Comment by getdatakick** [comment]
> For both your questions above, my view is yes and yes. With sane defaults, I can't see anything that goes markedly wrong.

Again, I don't agree with this. That would mean, for example, that you couldn't set your shop to send Order confirmation email but not Order Status Changed email. You could either send both, or none. 

Also, this could lead to weird situations when, for some payment methods, the Order confirmation email would be sent, and for other methods it wouldn't. That's not right. Order confirmation email should be send for every order (this is even legally required in some countries).

Order confirmation email preferences should not be driven by Order Status settings. 

> 1. how about having a preference on Orders page that allows switching off pdf attachment for order confirmation?
> 2. how about offloading this option to mail alerts module?

Yes, both could be done. I personally don't like #1 as it's a one-off solution for this particular use case, and not a systematic fix... but it's easy to do, of course

**Comment by NaturePinks** [comment]
Thanks, you are right the order status based attachment may be restrictive where the merchant expects a different behavior for confirmation and status change.

**Comment by x97wehner** [comment]
Resurfacing this as a needed enhancement @getdatakick . Additional reasoning:
- Attaching anything to an outbound email from a shop drives down the shop's overall email deliverability score long-term. For this reason it's not done competitive online store platforms anymore without some additional B2B setup. 
- A shop that services both consumers as well as businesses needs the ability to turn off invoices for consumers based upon order status. They most likely will still need for B2B customers. Each of these can be handled correctly with order statuses for everything else, but the inability to turn off the invoice email attachment for confirmations remains an issue.
